### PR TITLE
Implement flat shading

### PR DESCRIPTION
**Pull request by almarklein**
Flat shading is nice if you want lighting but also want to see your triangle. Or if you want a retro look.

image

### CODE REVIEW

File: pygfx/renderers/wgpu/meshrender.py [hunk 1]
```diff
@@ -488,6 +491,29 @@ def helpers(self):
...
+
+            let u = dpdx(world_pos);
+            let v = dpdy(world_pos);
+            var normal = normalize(cross(u, v));
+
+            // The normal calculated above may not be oriented correctly.
+            // We have two flags: is_front and u_stdinfo.flipped_winding.
+            // Note that lighting_phong() also applies the is_front flag.
+            // Below code means: flip the normal if the XOR of these two flags is true.
+            normal = select(normal, -normal, (select(0, 1, is_front) + u_stdinfo.flipped_winding) == 1);
```
Comment (almarklein) [inline pygfx/renderers/wgpu/meshrender.py:511]: I don't have a good way to explain this, but I tried this on the validate_culling.py example (using the flat material instead), and this is what I found makes it work :)

**Comment by almarklein** [comment]
Mmm, in ThreeJS the MeshPhongMaterial has a `.flatShading` property. It feels like a different material to me, what do you think @Korijn ?

**Comment by Korijn** [comment]
LGTM!

> Or if you want a retro look

It can look absolute gorgeous. You can also makes things look out of place/jarring by having a flat shaded object in an otherwise phong shaded scene.

**Comment by Korijn** [comment]
> Mmm, in ThreeJS the MeshPhongMaterial has a `.flatShading` property. It feels like a different material to me, what do you think @Korijn ?

I don't think it matters much :) I'm fine with having it be a separate material.

**Pull request merged**